haskell/private/cabal_wrapper.sh.tpl

```diff
@@ -81,6 +81,29 @@ while [[ $1 != -- ]]; do
 done
 shift 1
 
+declare -a include_dirs
+declare -a path_args
+for arg in $@; do
+    if [[ $arg =~ --extra-include-dirs=(.*) ]]; then
+      include_dirs+=("${BASH_REMATCH[1]}")
+    else
+      path_args+=("$arg")
+    fi
+done
+
+# Create one directory with a link to all the possibly needed header files.
+# This is required to avoid blowing-up the maximum command-line size when
+# there are too many include dirs
+local_include_dir="$execroot/local-package-db"
+mkdir "$local_include_dir"
+chmod +w $local_include_dir
+for db in "${include_dirs[@]}"; do
+    for conf_file in $db/*; do
+        ln -s "$(realpath $conf_file)" $local_include_dir/
```

Comment:
Can't this be solved by simply ignoring the error or overwriting the previous one ? I assume that if there's a collision in the available includes then one of them must take precedence (though I don't know whether it is specified somewhere which one should be taken into account)

Reply:
I think it would be more like a union of the directory trees. I think in this example `.` would contain source header files and `bazel-out/k8-fastbuild/bin` generated headers. In a sequence of `-I` flags it would be the left-biased union of the directory trees. Things get more complicated when `-iquote`, `-isystem`, and `-idirafter` get involved, but that shouldn't matter for `--extra-include-dirs`. AFAIK those all translate to `-I`.